security/acme-client/src/opnsense/mvc/app/library/OPNsense/AcmeClient/LeValidation/DnsTransip.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * Copyright (C) 2021 Nim G
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\AcmeClient\LeValidation;
+
+use OPNsense\AcmeClient\LeValidationInterface;
+use OPNsense\Core\Config;
+
+/**
+ * Transip API
+ * @package OPNsense\AcmeClient
+ */
+class DnsTransip extends Base implements LeValidationInterface
+{
+    public function prepare()
+    {
+        $this->acme_env['TRANSIP_Token'] = (string)$this->config->dns_transip_token;
```

Comment:
@MeganerdNL Looking at the documentation [here](https://github.com/acmesh-official/acme.sh/wiki/dnsapi#106-use-transip-domain-api) and the [source code](https://github.com/acmesh-official/acme.sh/blob/dev/dnsapi/dns_transip.sh), it seems to me that this will not work. The environment variable "TRANSIP_Token" is not used anywhere. What makes you think that this would work?

Reply:
Ok I see. I just copied from another entry with a token. I could not find an entry for that either. I think I messed up with that. I thought it was oke, but apparently it does not. I am not able to resolve this then by myself I'm afraid.

Reply:
I think I see where I got it wrong now what variable I should use. TRANSIP_Key_File. I can look into it tomorrow when I'm at my pc.